website/content/docs/v0.14/Learn More/developing-talos.md

```diff
@@ -0,0 +1,225 @@
+
+If the installation flow needs to be tested, `--with-bootloader=false` shouldn't be used.
+
+## Console Logs
+
+Watching console logs is easy with `tail`:
+
+```bash
+tail -F ~/.talos/clusters/talos-default/talos-default-*.log
+```
+
+## Interacting with Talos
+
+Once `talosctl cluster create` finishes successfully, `talosconfig` and `kubeconfig` will be set up automatically to point to your cluster.
+
+Start playing with `talosctl`:
+
+```bash
+talosctl -n 172.20.0.2 version
+talosctl -n 172.20.0.3,172.20.0.4 dashboard
+talosctl -n 172.20.0.4 get members
+```
+
+Same with `kubectl`:
+
+```bash
+kubectl get nodes -o wide
+```
+
+You can deploy some Kubernetes workloads to the cluster.
+
+You can edit machine config on the fly with `talosctl edit mc --immediate`, config patches can be applied via `--config-patch` flags, also many features have specific flags in `talosctl cluster create`.
+
+## Quick Reboot
+
+To reboot whole cluster quickly (e.g. to pick up a change made in the code):
+
+```bash
+for socket in ~/.talos/clusters/talos-default/talos-default-*.monitor; echo "q" | sudo socat - unix-connect:$socket; end
+```
+
+Sending `q` to a single socket allows to reboot a single node.
+
+> Note: This command performs immediate reboot (as if the machine was powered down and immediately powered back up), for normal Talos reboot use `talosctl reboot`.
+
+## Development Cycle
+
+Fast development cycle:
+
+- bring up a cluster
+- make code changes
+- rebuild `initramfs` with `make initramfs`
+- reboot a node to pick new `initramfs`
+- verify code changes
+- more code changes...
+
+Some aspects of Talos development require to enable bootloader (when working on `installer` itself), in that case quick development cycle is no longer possible, and cluster should be destroyed and recreated each time.
+
+## Running Integration Tests
+
+If integration tests were changed (or when running them for the first time), first rebuild the integration test binary:
+
+```bash
+rm -f  _out/integration-test-linux-amd64; make _out/integration-test-linux-amd64
+```
+
+Running short tests against QEMU provisioned cluster:
+
+```bash
+_out/integration-test-linux-amd64 \
+    -talos.provisioner=qemu \
+    -test.v \
+    -talos.crashdump=false \
+    -test.short \
+    -talos.talosctlpath=$PWD/_out/talosctl-linux-amd64
+```
+
+Whole test suite can be run removing `-test.short` flag.
+
+Specfic tests can be run with `-test.run=TestIntegration/api.ResetSuite`.
+
+## Build Flavors
+
+`make <something> WITH_RACE=1` enables Go race detector, Talos runs slower and uses more memory, but memory races are detected.
+
+`make <something> WITH_DEBUG=1` enables Go profiling and other debug features, useful for local development.
+
+## Destroying Custer
+
+```bash
+sudo -E ../talos/_out/talosctl-linux-amd64 cluster destroy --provisioner=qemu
+```
```

Comment:
worth mentioning that if you rebooted the machine, you need to manualy cleanup `~/.talos/clusters/<name>` dir

Reply:
thanks, I'll add that